src/args.rs

```diff
@@ -1,173 +1,106 @@
-use crate::auth;
-use crate::listing;
-use clap::{crate_authors, crate_description, crate_name, crate_version};
 use std::net::{IpAddr, Ipv4Addr, Ipv6Addr};
 use std::path::PathBuf;
+use structopt::StructOpt;
+
+use crate::auth;
+use crate::listing;
 
 /// Possible characters for random routes
 const ROUTE_ALPHABET: [char; 16] = [
     '1', '2', '3', '4', '5', '6', '7', '8', '9', '0', 'a', 'b', 'c', 'd', 'e', 'f',
 ];
 
-/// Checks wether a path is valid, i.e. it exists on the system and points to a file/directory
-fn is_valid_path(path: String) -> Result<(), String> {
-    let path_to_check = PathBuf::from(path);
-    if path_to_check.is_file() || path_to_check.is_dir() {
-        return Ok(());
-    }
-    Err(String::from(
-        "Path either doesn't exist or is not a regular file or a directory",
-    ))
-}
-
-/// Checks wether a port is valid
-fn is_valid_port(port: String) -> Result<(), String> {
-    port.parse::<u16>()
-        .and(Ok(()))
-        .or_else(|e| Err(e.to_string()))
+#[derive(StructOpt, Debug)]
+#[structopt(name = "miniserve")]
+struct CLIArgs {
+    /// Be verbose, includes emitting access logs
+    #[structopt(short = "v", long = "verbose")]
+    verbose: bool,
+    /// Which path to serve
```

Comment:
Let's space the arguments vertically to make them neater to read. Currently, this is like a big block. Let's have
```
/// docstring
#[structopt(stuff)]
verbose: bool,

/// docstring
...
```